api/observation.go

```diff
@@ -30,16 +32,47 @@ const (
 	getObservationsAction = "getObservations"
 )
 
+var (
+	observationNotFound = map[error]bool{
+		errs.ErrDatasetNotFound:      true,
+		errs.ErrEditionNotFound:      true,
+		errs.ErrVersionNotFound:      true,
+		errs.ErrObservationsNotFound: true,
+	}
+
+	observationBadRequest = map[error]bool{
+		errs.ErrTooManyWildcards: true,
+	}
+)
+
+type observationQueryError struct {
+	message string
+	status  int
+}
+
+func (e observationQueryError) Error() string {
+	return e.message
+}
+
 func errorIncorrectQueryParameters(params []string) error {
-	return fmt.Errorf("Incorrect selection of query parameters: %v, these dimensions do not exist for this version of the dataset", params)
+	return observationQueryError{
+		message: fmt.Sprintf("Incorrect selection of query parameters: %v, these dimensions do not exist for this version of the dataset", params),
+		status:  http.StatusBadRequest,
```

Comment:
It looks like all instances of `observationQueryError` result in a `http.StatusBadRequest`. If this is the case consider removing the status code from the error type. Then in the `handleObservationsErrorType` function just set the response to  `http.StatusBadRequest` if the type is `observationQueryError`.